plugins/zenoh-plugin-storage-manager/src/replica/digest.rs

```diff
@@ -683,21 +701,18 @@ impl Digest {
 
 // functions for alignment
 impl Digest {
-    // return mismatching eras
-    pub fn get_era_diff(&self, other: HashMap<EraType, Interval>) -> HashSet<EraType> {
-        let mut result = HashSet::new();
-        for era in vec![EraType::Hot, EraType::Warm, EraType::Cold] {
-            if other.contains_key(&era) && other.get(&era).unwrap().checksum != 0 {
-                if self.eras.contains_key(&era) {
-                    if self.eras.get(&era).unwrap().checksum != other.get(&era).unwrap().checksum {
-                        result.insert(era);
-                    }
-                } else {
-                    result.insert(era);
+    // check of the other era has more content
+    pub fn era_has_diff(&self, era: &EraType, other: &HashMap<EraType, Interval>) -> bool {
+        if other.contains_key(era) {
```

Comment:
This code could be more clear as such:
```rust
match (other.get(era), self.eras.get(era)) {
    (Some(other_interval), Some(my_interval)) => other_interval != my_interval,
    (Some(_), None) => true,
    _ => false
}
```


BTW, if `self` contains the `era` and `other` doesn't, isn't it a diff ?

Reply:
> if `self` contains the era and `other` doesn't, isn't it a diff ?

It is a diff, but as far as this replica is concerned (who has the `self`), it is not missing anything, it is upto the replica that sent the `other` digest to process it.